### PR TITLE
add mirage-net{-lwt}.1.2.0

### PR DESCRIPTION
**Pull request by hannesm**
for ipaddr 3.0.0 changes (macaddr now a separate opam package) fixes uninstallable packages (from #13240)

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
There is a missing dependency
```
#=== ERROR while compiling mirage-net-lwt.1.2.0 ===============================#
# context              2.0.2 | linux/x86_64 | ocaml-base-compiler.4.07.1 | pinned(https://github.com/mirage/mirage-net/releases/download/v1.2.0/mirage-net-1.2.0.tbz)
# path                 ~/.opam/4.07/.opam-switch/build/mirage-net-lwt.1.2.0
# command              ~/.opam/4.07/bin/jbuilder build -p mirage-net-lwt -j 127
# exit-code            1
# env-file             ~/.opam/log/mirage-net-lwt-20-d14459.env
# output-file          ~/.opam/log/mirage-net-lwt-20-d14459.out
### output ###
# File "lwt/jbuild", line 4, characters 16-26:
# 4 |   (libraries   (mirage-net lwt cstruct io-page macaddr))
#                     ^^^^^^^^^^
# Error: Library "mirage-net" not found.
# Hint: try: dune external-lib-deps --missing -p mirage-net-lwt @install

- File "lwt/jbuild", line 4, characters 16-26:
- 4 |   (libraries   (mirage-net lwt cstruct io-page macaddr))
-                     ^^^^^^^^^^
- Error: Library "mirage-net" not found.
- Hint: try: dune external-lib-deps --missing -p mirage-net-lwt @install
```

File: packages/mirage-net-lwt/mirage-net-lwt.1.2.0/opam [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+dev-repo:      "git+https://github.com/mirage/mirage-net.git"
+doc:           "https://mirage.github.io/mirage-net/"
+authors:       ["Thomas Gazagnaire" "Anil Madhavapeddy" "Gabriel Radanne"
+               "Mindy Preston" "Thomas Leonard" "Nicolas Ojeda Bar"
+               "Dave Scott" "David Kaloper" "Hannes Mehnert" "Richard Mortier"]
+tags:          [ "org:mirage"]
+license:       "ISC"
+
+build: [
+  ["jbuilder" "subst"] {pinned}
```
Comment (kit-ty-kate) [inline packages/mirage-net-lwt/mirage-net-lwt.1.2.0/opam]: With jbuilder, subst requires `"-p" name`

File: packages/mirage-net/mirage-net.1.2.0/opam [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+dev-repo:      "git+https://github.com/mirage/mirage-net.git"
+doc:           "https://mirage.github.io/mirage-net/"
+authors:       ["Thomas Gazagnaire" "Anil Madhavapeddy" "Gabriel Radanne"
+               "Mindy Preston" "Thomas Leonard" "Nicolas Ojeda Bar"
+               "Dave Scott" "David Kaloper" "Hannes Mehnert" "Richard Mortier"]
+tags:          [ "org:mirage"]
+license:       "ISC"
+
+build: [
+  ["jbuilder" "subst"] {pinned}
```
Comment (kit-ty-kate) [inline packages/mirage-net/mirage-net.1.2.0/opam]: see previous comment

**Comment by hannesm** [comment]
@kit-ty-kate thx for the review, I just pushed fixes

**Comment by avsm** [comment]
could you merge this against opam-repo master so some of the macaddr problems can be fixed in the datakit-ci?

**Comment by hannesm** [comment]
@avsm I just force-pushed after rebasing on master here

**Comment by kit-ty-kate** [comment]
Thanks!

**Pull request merged**